docs/Solidity.g4

```diff
@@ -371,21 +371,27 @@ subAssembly
   : 'assembly' identifier assemblyBlock ;
 
 numberLiteral
-  : (DecimalNumber | HexNumber) NumberUnit? ;
+  : (DecimalNumberWithoutExponent | DecimalNumberWithExponent | HexNumber) NumberUnit? ;
 
 identifier
   : ('from' | 'calldata' | 'address' | Identifier) ;
 
 BooleanLiteral
   : 'true' | 'false' ;
 
-DecimalNumber
-  : ( DecimalDigits | (DecimalDigits? '.' DecimalDigits) ) ( [eE] '-'? DecimalDigits )? ;
+DecimalNumberWithoutExponent
```

Comment:
Would it make sense to at least reduce duplication here? We should have at least a parser rule `decimalNumber : DecimalNumberWithoutExponent | DecimalNumberWithExponent`.